adapters/airflow_adapter.py

```diff
@@ -0,0 +1,133 @@
+"""The Airflow Adapter."""
+from datetime import datetime
+import threading
+from adapter_abc import Adapter
+from from_sql import read_releases
+from from_sql import read_tasks
+from resources.filter_releases import filter_releases
+from resources.filter_releases import sort
+from to_sql import to_sql_releases
+from to_sql import to_sql_release
+from to_sql import to_sql_task
+from to_sql import to_sql_tasks
+import logging
+
+CACHE_TTL = 1800
+
+class AirflowAdapter(Adapter):
+  """Provides a way to interact with the Airflow Database, and get data from it."""
+
+  def __init__(self, airflow_db):
+    self._airflow_db = airflow_db
+    self._lock = threading.Lock()
+    self._release_types = set()
+    self._branches = set()
+    self._cache_last_updated = datetime.fromtimestamp(0)
+    self._update_cache()
+
+  def get_releases(self, filter_options):
+    """Gets all releases fitting filter_options.
+
+    Args:
+      filter_options: object containing all filter parameters
+
+    Returns:
+      releases_data: a filtered, sorted list of release objects
+    """
+    # build the SQL query
+    release_query = to_sql_releases(filter_options)
+    # get the data from SQL
+    release_data = self._airflow_db.query(release_query)
+    # package the data into release objects with all neccessary info
+    releases_data = read_releases(release_data, self._airflow_db)
+    # filter for the stuff that SQL can't natively filter for,
+    # due to some data being based on a tasks
+    releases_data = filter_releases(releases_data, filter_options)
+    # sort with the stuff that SQL can't natively sort according to,
+    # due to some data being based on tasks
+    releases_data = sort(releases_data, filter_options)
+
+    return releases_data
+
+  def get_release(self, release_name):
+    # construct SQL query
+    release_query = to_sql_release(release_name)
+    # get data from SQL
+    release_data = self._airflow_db.query(release_query)
+    # package data into a release object
+    release_data = read_releases(release_data, self._airflow_db)
+
+    return release_data
+
+  def get_tasks(self, dag_id, execution_date):
```

Comment:
I am thinking that execution_data is really airflow implementation and we should not bring that the interface, can we use release_id instead, and do a mapping inside the AirflowAdapter impl ?

Reply:
Sure I'll make the change